basetest.pm

```diff
@@ -320,6 +320,11 @@ sub run_post_fail {
         eval { $self->post_fail_hook; };
         bmwqemu::diag("post_fail_hook failed: $@") if $@;
         $self->{post_fail_hook_running} = 0;
+
+        # There might be more messages on serial now.
+        # Update the offset to not stumble upon them in the next module.
+        my $json = $self->get_serial_output_json($serial_file_pos);
+        $serial_file_pos = $json->{position};
```

Comment:
It doesn't look like we're ever calling `get_serial_output_json` without this assignment. So maybe move `$serial_file_pos = $json->{position};` into that function?

Reply:
I changed `get_serial_output_json` a bit, how's the new approach? VR is updated.

Reply:
Looks good.

Reply:
https://github.com/os-autoinst/os-autoinst/runs/4117429312?check_suite_focus=true#step:3:581 says "basetest::get_serial_output_json does not exist! at ./17-basetest.t line 69."

Reply:
Fixed